src/extension.ts

```diff
@@ -5,62 +5,122 @@ import * as net from 'net';
 import * as cp from 'child_process';
 import * as fs from 'fs';
 import * as path from 'path';
+import * as os from 'os';
+import * as getPort from 'get-port';
 
 import { workspace, ExtensionContext, Disposable, window, Uri, commands, OutputChannel } from 'vscode';
-import { RevealOutputChannelOn, LanguageClient, LanguageClientOptions, ServerOptions, TransportKind, Executable, ErrorHandler, Message, ErrorAction, CloseAction } from 'vscode-languageclient';
-
-export function activate(context: ExtensionContext) {
-
-	//let serverOptions: ServerOptions = {command:'python3', args:['-m', 'inmanta-lsp']};
-
-	function startTCP() {
-		// have to start server by hand, debugging only
-		const serverOptions: ServerOptions = function () {
-			return new Promise((resolve, reject) => {
-				const client = new net.Socket();
-				client.connect(5432, "127.0.0.1", function () {
-					resolve({
-						reader: client,
-						writer: client
-					});
-				});
-			});
-		};
+import { RevealOutputChannelOn, LanguageClient, LanguageClientOptions, ServerOptions, Executable, ErrorHandler, Message, ErrorAction, CloseAction } from 'vscode-languageclient';
+
+
+export async function activate(context: ExtensionContext) {
+	let lsOutputChannel = null;
+
+	async function startServerAndClient() {
+		const clientOptions = getClientOptions();
+		if (os.platform() === "win32") {
+			return await startTcp(clientOptions);
+		} else {
+			return startPipe(clientOptions);
+		}
+	}
+
+	function getClientOptions(): LanguageClientOptions {
+		const errorhandler = new LsErrorHandler();
 
 		// Options to control the language client
+		const compilerVenv: string = workspace.getConfiguration('inmanta').compilerVenv || Uri.joinPath(context.storageUri, ".env-ls-compiler").fsPath;
 		const clientOptions: LanguageClientOptions = {
 			// Register the server for inmanta documents
-			documentSelector: [{ scheme: 'file', language: 'inmanta' }]
+			documentSelector: [{ scheme: 'file', language: 'inmanta' }],
+			errorHandler: errorhandler,
+			revealOutputChannelOn: RevealOutputChannelOn.Info,
+			initializationOptions: {
+				compilerVenv: compilerVenv
+			}
 		};
+		return clientOptions;
+	}
+	
+	async function startTcp(clientOptions: LanguageClientOptions) {
+		const host = "localhost";
```

Comment:
Wouldn't it be better to use `127.0.0.1` instead of `localhost`. In the past we had issues on core, because localhost must be translated to an address. This might be an IPv4 or an IPv6 address, but you never know which. In this case three independent libraries will do that translation:
* the `getPort()` call,
* the `tcpServer()` and
* the `net.connect()`

If one of those libraries translate localhost in a different way, there is an issue. @wouterdb What do you think?